models/quality_measures/intermediate/nqf2372_breast_cancer_screening/quality_measures__int_nqf2372_exclude_mastectomy.sql

```diff
@@ -169,9 +171,102 @@ with denominator as (
 
 )
 
+/*
+    Women who had a bilateral mastectomy or who have a history of a bilateral
+    mastectomy
+*/
+, bilateral_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+          'Bilateral Mastectomy'
+        , 'History of bilateral mastectomy'
+    )
+
+)
+
+, right_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+          'Status Post Right Mastectomy'
+        , 'Unilateral Mastectomy Right'
+    )
+
+)
+
+, left_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+          'Status Post Left Mastectomy'
+        , 'Unilateral Mastectomy Left'
+    )
+
+)
+
+, unspecified_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+        'Unilateral Mastectomy, Unspecified Laterality'
+    )
+
+)
+
+/*
+    Women for whom there is evidence of a right AND a left unilateral mastectomy
+    or unspecific mastectomies on different dates
+*/
+, unilateral_mastectomy as (
```

Comment:
Added new logic for unilateral mastectomy. Must have evidence of both left and right mastectomies.